OpenEdXMobile/src/main/java/org/edx/mobile/view/adapters/CourseDiscussionResponsesAdapter.java

```diff
@@ -358,21 +373,25 @@ protected void onResponse(@NonNull final DiscussionComment comment) {
         }
     }
 
-    private void bindSocialView(DiscussionSocialLayoutViewHolder holder, final int position, final DiscussionComment response) {
+    private void bindSocialView(final DiscussionSocialLayoutViewHolder holder, final int position, final DiscussionComment response) {
         holder.setDiscussionResponse(response);
 
         holder.voteViewContainer.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
 
+                boolean isVoted = holder.toggleVote(discussionThread.isVoted() ? discussionThread.getVoteCount()-1: discussionThread.getVoteCount());
```

Comment:
Ooops!!! Bug spotted 🐞

Hint: `discussionThread` isn't the right object to use here.

Reply:
I am not entirely sure what direction you would prefer here.  I can change the 

"discussionThread.isVoted()"
to 
holder.voteViewContainer.isSelected()

But it would take a lot of string manipulation to use the holder for the integer (unless you think its worth it).  

I could also use thread and make it a final variable?

Reply:
`discussionThread` is the wrong object to use altogether. Replacing it with `response` variable would solve the problem is what i meant.

Reply:
@miankhalid  
their is no variable response.  The whole point of my logic is that it runs before the api call is made.

Reply:
Check line # 376, `response` is being passed as an arg to the `bindSocialView` function.

```java
private void bindSocialView(final DiscussionSocialLayoutViewHolder holder, final int position, final DiscussionComment response) {
```

Reply:
@miankhalid  

Thanks i was looking at the wrong section, i pushed an update is that better?

Reply:
Lemme check.